x/gamm/keeper/pool_twap.go

```diff
@@ -182,6 +183,7 @@ func (k Keeper) GetRecentPoolTwapSpotPrice(
 		return sdk.Dec{}, fmt.Errorf("pool twap history prior to current time does not exist")
 	}
 
+	// TODO: use duration.Seconds?
 	desiredTime := ctx.BlockTime().Add(-duration * time.Second)
 	desiredTimeAdjacentPoolTwap, exists := k.GetPoolTwapHistory(ctx, poolId, desiredTime.Add(time.Second))
```

Comment:
Can you add some comments explaining desired time and why this mysterious time.second is added?

Should it be time.nanosecond or smth added?

Reply:
Sure! Definitely should make this clearer